### PR TITLE
qemu-guest-agent: fix guest-exec issue

### PR DESCRIPTION
**Pull request by xiagao**
The behavior of linux vm and windows vm is different
 when run guest-exec cmd with capture_output is true.

id: 1745887
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
(1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_exec.q35: PASS (64.53 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_exec.q35: PASS (106.71 s)

**Comment by xiagao** [comment]
@xiangchunfu  call for your review... thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 1]
```diff
@@ -1299,21 +1299,42 @@ def _guest_cmd_run(guest_cmd, cmd_args=None, env_qga=None,
             if not result["exited"]:
                 test.error("Guest cmd is still running, pls login guest to"
                            " handle it or extend your timeout.")
-            # check the exitcode and output/error data result
-            if result["exitcode"] == 0:
+            # check the exitcode and output/error data if capture_output
+            #  is true
+            if capture_output is not True:
+                return result
```
Comment (xiangchunfu) [inline qemu/tests/qemu_guest_agent.py]: @xiagao  capture_output is not True,  only return result.   If have some error message in result.  maybe you cann't find it.  And case test result is Pass.   We don't need to deal with 'result'?  only return it.

File: qemu/tests/qemu_guest_agent.py [hunk 2]
```diff
@@ -1299,21 +1299,42 @@ def _guest_cmd_run(guest_cmd, cmd_args=None, env_qga=None,
...
-            else:
-                if "out-data" in result:
-                    test.fail("When exitcode is 1, should not get output data.")
-                elif "err-data" in result:
-                    err_data = base64.b64decode(result["err-data"]).decode()
-                    logging.info("The guest cmd failed,"
-                                 "the error info is: \n %s" % err_data)
+                                 "the output is:\n%s." % out_data)
+                else:
+                    test.fail("There is no output with capture_output is true.")
```
Comment (xiangchunfu) [inline qemu/tests/qemu_guest_agent.py]: @xiagao  For windows guest.  You don't  confirm err-data in result, why?

File: qemu/tests/qemu_guest_agent.py (hunk 1 above)
Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: @xiangchunfu 
1. the result is output.
2. this code is a sub function, only need return the result.

File: qemu/tests/qemu_guest_agent.py (hunk 2 above)
Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: @xiangchunfu as there is only out-data for windows guest.

**Comment by xiangchunfu** [comment]
Acked-by: Xiangchun Fu xfu@redhat.com

**Comment by xiagao** [comment]
@vivianQizhu Could you help review? Thanks.

**Comment by vivianQizhu** [comment]
@xiagao Is the id number correct?

**Comment by xiagao** [comment]
Sorry, it's wrong, update now.

**Comment by vivianQizhu** [comment]
Ack

**Pull request merged**